### PR TITLE
461 audit fix 9

### LINKED ISSUE #461: Audit fix 09

### PR DESCRIPTION
**Pull request by JohnGuilding**
## What is this PR doing?
Implements audit fix 9

## How can these changes be manually tested?
- run contract tests

## Does this PR resolve or contribute to any issues?
contributes to #461 

## Checklist
- [x] I have manually tested these changes
- [x] Post a link to the PR in the group chat

## Guidelines
- If your PR is not ready, mark it as a draft
- The `resolve conversation` button is for reviewers, not authors
  - (But add a 'done' comment or similar)

### CODE REVIEW

File: contracts/test/recovery-test.ts [hunk 1]
```diff
@@ -25,6 +29,7 @@ const signWalletAddress = async (
 
 describe("Recovery", async function () {
   const safetyDelaySeconds = 7 * 24 * 60 * 60;
+  const signatureExpiryWindow = 1200;
```
Comment (JohnGuilding) [inline contracts/test/recovery-test.ts]: Chose this arbitrarily. Is equal to 100 12 second blocks.

Was a bit torn on whether this was a good variable name. Suggestions welcome :)

**Comment by JohnGuilding** [comment]
CI seems to be failing due to some typescript issues in the client module. Doesn't seem related to this PR - got the same failures running clients tests in latest from `contract-updates`. Can look into this if needed but leaving for now.

File: contracts/test/recovery-test.ts (hunk 1 above)
Comment (jzaki) [reply contracts/test/recovery-test.ts]: It looks to be used as an offset, so `signatureExpiryOffsetSeconds`? (I like putting units in where meaningful)

File: contracts/test/recovery-test.ts [hunk 2]
```diff
@@ -581,4 +627,116 @@ describe("Recovery", async function () {
...
+    await fx.call(
+      wallet1,
+      vg,
+      "setBLSKeyForWallet",
+      [addressSignature, wallet2.PublicKey(), invalidSignatureExpiryTimestamp],
+      1,
+      30_000_000,
+    );
+
+    await fx.advanceTimeBy(safetyDelaySeconds + 1);
```
Comment (jzaki) [inline contracts/test/recovery-test.ts:669]: Minor: a couple of comments around the expiry timestamp, and expectations that the first `set..` will succeed.

**Review by jzaki — APPROVED** [review]

File: contracts/test/recovery-test.ts (hunk 1 above)
Comment (JohnGuilding) [reply contracts/test/recovery-test.ts]: Like that - agree the units make sense here. Will update to `signatureExpiryOffsetSeconds`

File: contracts/test/recovery-test.ts (hunk 2 above)
Comment (JohnGuilding) [reply contracts/test/recovery-test.ts:669]: Added

**Pull request merged**